tiny_dnn/layers/layer.h

```diff
@@ -704,6 +867,10 @@ class layer : public node {
       }
       resize(ith_out_node(i)->get_gradient());
     }
+
+    for (size_t i = 0; i < parameters_.size(); i++) {
+      parameters_[i]->resize_grad(sample_count);
+    }
```

Comment:
now I see what you meant before. Here you might want to resize or create a new tensor with size NxD1xD2xDn, where N=sample_count. `resize_grad` calls reshape, right? Not sure just ideas :smile: